test/extensions/common/wasm/wasm_test.cc

```diff
@@ -0,0 +1,299 @@
+  }
+  MOCK_METHOD2(scriptLog_, void(spdlog::level::level_enum level, absl::string_view message));
+};
+
+class WasmCommonTest : public testing::TestWithParam<std::string> {};
+
+INSTANTIATE_TEST_SUITE_P(Runtimes, WasmCommonTest, testing::Values("v8", "null"));
+
+TEST_P(WasmCommonTest, Logging) {
+  Stats::IsolatedStoreImpl stats_store;
+  Api::ApiPtr api = Api::createApiForTest(stats_store);
+  Upstream::MockClusterManager cluster_manager;
+  Event::DispatcherPtr dispatcher(api->allocateDispatcher());
+  auto scope = Stats::ScopeSharedPtr(stats_store.createScope("wasm."));
+  NiceMock<LocalInfo::MockLocalInfo> local_info;
+  auto name = "";
+  auto root_id = "";
+  auto vm_id = "";
+  auto vm_configuration = "logging";
+  auto plugin = std::make_shared<Extensions::Common::Wasm::Plugin>(
+      name, root_id, vm_id, envoy::config::core::v3alpha::TrafficDirection::UNSPECIFIED, local_info,
+      nullptr);
+  auto wasm = std::make_shared<Extensions::Common::Wasm::Wasm>(
+      absl::StrCat("envoy.wasm.runtime.", GetParam()), vm_id, vm_configuration, scope,
+      cluster_manager, *dispatcher);
+  EXPECT_NE(wasm, nullptr);
+  auto wasm_weak = std::weak_ptr<Extensions::Common::Wasm::Wasm>(wasm);
+  auto wasm_handler = std::make_unique<Extensions::Common::Wasm::WasmHandle>(std::move(wasm));
+  std::string code;
+  if (GetParam() == "v8") {
+    code = TestEnvironment::readFileToStringForTest(TestEnvironment::substitute(
+        absl::StrCat("{{ test_rundir }}/test/extensions/common/wasm/test_data/test_cpp.wasm")));
+  } else {
+    // The name of the Null VM plugin.
+    code = "CommonWasmTestCpp";
+  }
+  EXPECT_FALSE(code.empty());
+  auto context = std::make_unique<TestContext>(wasm_weak.lock().get());
+
+  EXPECT_CALL(*context, scriptLog_(spdlog::level::trace, Eq("test trace logging")));
+  EXPECT_CALL(*context, scriptLog_(spdlog::level::debug, Eq("test debug logging")));
+  EXPECT_CALL(*context, scriptLog_(spdlog::level::warn, Eq("test warn logging")));
+  EXPECT_CALL(*context, scriptLog_(spdlog::level::err, Eq("test error logging")));
+  EXPECT_CALL(*context, scriptLog_(spdlog::level::info, Eq("on_configuration configure-test")));
+  EXPECT_CALL(*context, scriptLog_(spdlog::level::info, Eq("on_done logging")));
+  EXPECT_CALL(*context, scriptLog_(spdlog::level::info, Eq("on_delete logging")));
+
+  EXPECT_TRUE(wasm_weak.lock()->initialize(code, false));
+  wasm_weak.lock()->setContext(context.get());
+  auto root_context = context.get();
+  wasm_weak.lock()->startForTesting(std::move(context), plugin);
+  wasm_weak.lock()->configure(root_context, plugin, "configure-test");
+  wasm_handler.reset();
+  dispatcher->run(Event::Dispatcher::RunType::NonBlock);
+  // This will SEGV on nullptr if wasm has been deleted.
+  wasm_weak.lock()->configure(root_context, plugin, "done");
+  dispatcher->run(Event::Dispatcher::RunType::NonBlock);
+  dispatcher->clearDeferredDeleteList();
+}
+
+TEST_P(WasmCommonTest, BadSignature) {
+  if (GetParam() != "v8") {
```

Comment:
This should be:
```
if (GetParam() == "null") {
  return;
}
```
(same in other places)

Reply:
Done